src/main/java/com/github/discvrseq/walkers/VcfToLuceneIndexer.java

```diff
@@ -240,39 +243,75 @@ else if (line.getCountType() == VCFHeaderLineCount.INTEGER || line.getCountType(
 
                 // Add standard fields
                 doc.add(new TextField("contig", variant.getContig(), Field.Store.YES));
+                doc.add(new SortedDocValuesField("contig", new BytesRef(variant.getContig())));
+
                 doc.add(new TextField("ref", variant.getReference().getDisplayString(), Field.Store.YES));
+                doc.add(new SortedDocValuesField("ref", new BytesRef(variant.getReference().getDisplayString())));
+
                 doc.add(new TextField("alt", alt.getDisplayString(), Field.Store.YES));
+                doc.add(new SortedDocValuesField("alt", new BytesRef(alt.getDisplayString())));
 
                 doc.add(new IntPoint("start", variant.getStart()));
                 doc.add(new StoredField("start", variant.getStart()));
+                doc.add(new NumericDocValuesField("start", variant.getStart()));
 
                 doc.add(new IntPoint("end", variant.getEnd()));
                 doc.add(new StoredField("end", variant.getEnd()));
+                doc.add(new NumericDocValuesField("end", variant.getEnd()));
 
                 final int genomicPosition = getGenomicPosition(variant.getContig(), variant.getStart());
                 doc.add(new IntPoint("genomicPosition", genomicPosition));
                 doc.add(new StoredField("genomicPosition", genomicPosition));
-                doc.add(new SortedNumericDocValuesField("genomicPosition", genomicPosition));
+                doc.add(new NumericDocValuesField("genomicPosition", genomicPosition));
 
                 if (variant.hasGenotypes()) {
-                    variant.getGenotypes().stream().filter(g -> !g.isFiltered() && !g.isNoCall() && g.getAlleles().contains(alt)).map(Genotype::getSampleName).sorted().forEach(sample -> doc.add(new TextField("variableSamples", sample, Field.Store.YES)));
-                    variant.getGenotypes().stream().filter(g -> !g.isFiltered() && !g.isNoCall() && g.getAlleles().contains(alt) && g.isHomVar()).map(Genotype::getSampleName).sorted().forEach(sample -> doc.add(new TextField("homozygousVarSamples", sample, Field.Store.YES)));
+                    final String[] docValue = { null };
```

Comment:
I think this pattern is OK (at least I dont see anything saying it isnt), but be aware that AtomicReference<String> is another pattern to store a value outside a lambda.